app/gist/controller.js

```diff
@@ -86,6 +86,46 @@ export default Ember.Controller.extend({
     }
   },
 
+  createFile(filePath, fileProperties) {
+    if (filePath) {
+      if(this.get('model.files').findBy('filePath', filePath)) {
+        alert('A file with the name %@ already exists'.fmt(filePath));
+        return;
+      }
+
+      fileProperties.filePath = filePath;
+      let file = this.store.createRecord('gistFile', fileProperties);
+
+      this.get('model.files').pushObject(file);
+      this.notify.info('File %@ was added'.fmt(file.get('filePath')));
+      this.set('col1File', file);
+      this.set('activeEditorCol', '1');
+      this.send('contentsChanged');
+    }
+  },
+  /*
+   *  Test path is valid.  Presently only tests components are hyphenated.
+   */
+  hasInvalidPath(type, path){
+    let errorMsg = null;
+    if (type === 'component'){
+      let file = path.split('/').slice(-1)[0];
+      if (file.indexOf('.') > -1) {
+        errorMsg = "What part of WITHOUT FILE EXTENSION did you not understand ;-) ?";
```

Comment:
@rwjblue @joostdevries Do we want an error message at all or do we want to automatically strip the extension?